x/market/handler/endblock_test.go

```diff
@@ -0,0 +1,213 @@
+package handler
+
+import (
+	"errors"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ovrclk/akash/testutil"
+	"github.com/ovrclk/akash/x/market/types"
+)
+
+var (
+	// Create and seed the generator.
+	// Typically a non-fixed seed should be used, such as time.Now().UnixNano().
+	// Using a fixed seed will produce the same output on every run.
+	r = rand.New(rand.NewSource(550000))
+)
+
+type winnerTest struct {
+	desc      string
+	bids      []types.Bid
+	expWinner *types.Bid
+	expErr    error
+}
+
+func (w *winnerTest) testFunc(t *testing.T) {
+	winner, err := pickBidWinner(w.bids)
+	if !errors.Is(err, w.expErr) {
+		t.Errorf("returned err: %v does not match %v", err, w.expErr)
+	}
+	if w.expWinner != nil && !winner.Equals(w.expWinner.ID()) {
+		t.Errorf("unexpected winner: %#v\n%q : %v", winner, types.BidIDString(winner.BidID), winner.Price)
+		t.Logf("winner: %+v coin: %s", winner.ID(), winner.Price.String())
+	}
+}
+
+func TestBidWinner(t *testing.T) {
+	originOID := testutil.OrderID(t)
+
+	bid0 := createBid(t, originOID, int64(3))
+	bid1 := createBid(t, originOID, int64(5))
+	bid2 := createBid(t, originOID, int64(5))
+	bid3 := createBid(t, originOID, int64(4))
+	bid4 := createBid(t, originOID, int64(5))
+	bid5 := createBid(t, originOID, int64(5))
+	bid6 := createBid(t, originOID, int64(5))
+	bid7 := createBid(t, originOID, int64(8))
+	bid8 := createBid(t, originOID, int64(6))
+	bid9 := createBid(t, originOID, int64(5))
```

Comment:
again, would have been more convenient if `createBid()` took an `int` as its third argument.

Reply:
That's true. I'll change it.